packages/driver/cypress/integration/e2e/multi-domain/multi_domain_yield_spec.ts

```diff
@@ -0,0 +1,253 @@
+const { assertLogLength } = require('../../../support/utils')
+
+// @ts-ignore / session support is needed for visiting about:blank between tests
+describe('multi-domain', { experimentalSessionSupport: true, experimentalMultiDomain: true }, () => {
+  let logs: any = []
+
+  beforeEach(() => {
+    logs = []
+
+    cy.on('log:added', (attrs, log) => {
+      logs.push(log)
+    })
+
+    cy.visit('/fixtures/multi-domain.html')
+    cy.get('a[data-cy="multi-domain-secondary-link"]').click()
+  })
+
+  it('yields a value', () => {
+    cy.switchToDomain('foobar.com', () => {
+      cy
+      .get('[data-cy="dom-check"]')
+      .invoke('text')
+    }).should('equal', 'From a secondary domain')
+  })
+
+  it('yields the cy value even if a return is present', () => {
+    cy.switchToDomain('foobar.com', async () => {
+      cy
+      .get('[data-cy="dom-check"]')
+      .invoke('text')
+
+      const p = new Promise((resolve, reject) => {
+        setTimeout(() => {
+          resolve('text')
+        }, 50)
+      })
+
+      return p
+    }).should('equal', 'From a secondary domain')
+  })
+
+  it('errors if a cy command is present and it returns a sync value', (done) => {
+    cy.on('fail', (err) => {
+      assertLogLength(logs, 6)
+      expect(logs[5].get('error')).to.eq(err)
+      expect(err.message).to.include('`cy.switchToDomain()` failed because you are mixing up async and sync code.')
+
+      done()
+    })
+
+    cy.switchToDomain('foobar.com', () => {
+      cy
+      .get('[data-cy="dom-check"]')
+      .invoke('text')
+
+      return 'text'
+    })
+  })
+
+  it('yields synchronously', () => {
+    cy.switchToDomain('foobar.com', () => {
+      return 'From a secondary domain'
+    }).should('equal', 'From a secondary domain')
+  })
+
+  it('yields asynchronously', () => {
+    cy.switchToDomain('foobar.com', async () => {
+      return new Promise((resolve: (val: string) => any, reject) => {
+        setTimeout(() => {
+          resolve('From a secondary domain')
+        }, 1000)
+      })
+    }).should('equal', 'From a secondary domain')
+  })
+
+  it('succeeds if subject cannot be serialized and is not accessed synchronously', () => {
+    cy.switchToDomain('foobar.com', () => {
+      return {
+        symbol: Symbol(''),
+      }
+    }).then((obj) => {
+      return 'object not accessed'
+    }).should('equal', 'object not accessed')
+  })
+
+  it('throws if subject cannot be serialized and is accessed synchronously', (done) => {
+    cy.on('fail', (err) => {
+      assertLogLength(logs, 6)
+      expect(logs[5].get('error')).to.eq(err)
+      expect(err.message).to.include('`cy.switchToDomain()` could not serialize the subject due to one of its properties not being supported by the structured clone algorithm.')
```

Comment:
I wonder if this will be part of the `switchToDomain` documentation, in which case we can link to that.

Reply:
☝️ This was what i was going to suggest. Our doc can provide a buffer if the mdn page moves for whatever reason, instead of hard coding it into the package.

Reply:
That would be amazing. Sounds like a solid solution to me.